Code/RDGeneral/RDValue-taggedunion.h

```diff
@@ -404,12 +404,55 @@ inline int rdvalue_cast<int>(RDValue_cast_t v) {
   if (rdvalue_is<int>(v)) return v.value.i;
   throw boost::bad_any_cast();
 }
+
+template <>
+inline std::int8_t rdvalue_cast<std::int8_t>(RDValue_cast_t v) {
+  if (rdvalue_is<int>(v)) return static_cast<std::int8_t>(v.value.i);
+  throw boost::bad_any_cast();
+}
+
+template <>
+inline std::int16_t rdvalue_cast<std::int16_t>(RDValue_cast_t v) {
+  if (rdvalue_is<int>(v)) return static_cast<std::int16_t>(v.value.i);
+  throw boost::bad_any_cast();
+}
```

Comment:
I was a bit hesitant as well, but was following the lack of checking in static_cast

Reply:
Agreed, but we couldn't trigger these particular overflows before, so I think it's either worth keeping it that way (i.e. you can't do it) or adding the tests so that it only happens when there's no data loss